torch/_inductor/overrides.py

```diff
@@ -51,6 +59,12 @@ def replace_fx(gm: torch.fx.GraphModule):
                 and replacements[node.target] in replacements_using_triton_random
             ):
                 continue
+            if node.target in fallback_cpu_random and all(
+                example_input.device == torch.device("cpu")
+                for example_input in example_inputs
+                if isinstance(example_input, torch.Tensor)
```

Comment:
can you lift this cpu device check outside the loop just like `fuse_fx` does? we don't have to compute it multiple times in the loop.